proto-lens/src/Data/ProtoLens/Encoding/Parser.hs

```diff
@@ -104,6 +105,17 @@ withSized len message f = Parser $ \end pos ->
         else liftIO $ ParserResult pos' <$> f pos
 {-# INLINE withSized #-}
 
+-- | Run the given parsing action as if there are only 
+-- @len@ bytes remaining.  That is, once @len@ bytes have been
+-- consumed, 'atEnd' will return 'True' and other actions
+-- like 'getWord8' will act like there is no input remaining.
+isolate :: Int -> Parser a -> Parser a
```

Comment:
Where do we catch negative `len`?

Reply:
Good catch; fixed to fail the parse in that case.